### PR TITLE
helm 3: support writing multiple resources to the same file

### PR DESCRIPTION
**Pull request by torstenwalter**
**What this PR does / why we need it**:

Templates can contain multiple manifests. As I made a mistake while implementing `--output-dir` option in #5762  for `helm template` only the last resource is written to that file.

This PR fixes that.

**Special notes for your reviewer**:

**If applicable**:
- [ ] this PR contains documentation
- [x] this PR contains unit tests
- [ ] this PR has been tested for backwards compatibility

### CODE REVIEW

**Comment by bacongobbler** [comment]
Can you write some unit tests to cover this case?

**Comment by torstenwalter** [comment]
Where should these test be added?
`go test -timeout 30s helm.sh/helm/cmd/helm` is broken on dev-v3 branch. Are those executed?
Or should the unit tests be placed in `pkg/action/...`

**Comment by torstenwalter** [comment]
@bacongobbler Should be ready for review.

**Comment by hickeyma** [comment]
@torstenwalter To clarify here, this is to fix the issue whereby you have more than one resource in the same template file and the output just writes the last resource to the output manifest file? I am  asking because you mention "Templates can contain multiple manifests." in the description and it works for this scenario. I would think that this would also need to be fixed for v2.

File: pkg/action/install.go
Comment (hickeyma) [inline pkg/action/install.go]: You can drop the "else". Small 'linter' nit.

**Review by hickeyma — CHANGES_REQUESTED** [review]
Good catch @torstenwalter. Thanks for the fix. It is looking good and tests as expected on manual tests that I ran. Small fix needed.

File: pkg/action/install.go
Comment (torstenwalter) [reply pkg/action/install.go]: @hickeyma Can you change this directly? Otherwise I'll do it in a week when I am back at my desk.

**Comment by torstenwalter** [comment]
> @torstenwalter To clarify here, is this is to fix the issue whereby you have more than one resource in the same template file and the output just writes the last resource to the output manifest file? I am asking because you mention "Templates can contain multiple manifests." in the description and it does already work for this scenario.
> 
> I think that this would also need to be fixed for v2.

I think it works in v2.

File: pkg/action/install.go
Comment (hickeyma) [reply pkg/action/install.go]: Yiu can do it when you return @torstenwalter when you return. You have done all the hard work.

Comment (torstenwalter) [reply pkg/action/install.go]: @hickeyma Change is implemented.

**Review by hickeyma — APPROVED** [review]
Thanks @torstenwalter, LGTM. Can you squash the changes?

This will need another review before merge.

**Comment by torstenwalter** [comment]
@hickeyma squashed and rebased on dev-v3

**Comment by hickeyma** [comment]
> I think it works in v2.

Yes, it does.

File: internal/test/test.go
Comment (bacongobbler) [inline internal/test/test.go]: can you add a `t.Helper()` here? See above functions^

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Review by hickeyma — COMMENTED** [review]
@torstenwalter do you mind squashing the changes?

**Review by bacongobbler — APPROVED** [review]

**Comment by torstenwalter** [comment]
@hickeyma, @bacongobbler Thank you for considering this PR and reviewing it.

**Pull request merged**
